### ISSUE TITLE
Strange Unwanted Unsubscribe Issue.

### ISSUE THREAD

**Issue by ziming** [spatie/mailcoach-support#184]
Not sure if it's a bug on mailcoach side or an issue with my subscribers.

So currently my client has made 4 email campaigns so far with mailcoach. For some strange reasons there are 2 subscribers that always go unsubscribed right after each send.

Checking Campaign Unsubscribes doesn't show them. But when you go to email list section, the 2 subscribers will be marked unsubscribed with the same date time of the latest campaign send.

They don't appear in the bounced list of the campaign either.

When my client called the 2 subscribers, both of them insist they are still interested and want to receive the emails. 1 of them have also indicated that they have received the email so it wasn't a bounce. So for now after each campaign, my client re-subscribe these 2 emails back manually each time. But that's not a long term solution

What could possibly cause the issue of these 2 subscribers getting unsubscribed when they never wanted it?

TLDR:
- 2 subscribers always become unsubscribed after each campaign send
- It is not indicated in campaign unsubscribes (We have set up the SNS and ses feedback stuffs for all possible events we can check)
- The 2 subscribers did not click on any unsubscribe link to their knowledge.
- The emails to them are not bounced.
- Prior to this my client is using mailchimp and the 2 subscribers always stay subscribed.

If it helps, we are using a custom Mailable class with a custom Segment class. Since I read that there are still some issues with it right now.

Any help or possible causes will be great. Thank you.

**Comment by freekmurze** [comment]
I'm thinking this is not directly caused by mailcoach as manual unsubscribes, or bounces are the only way people get unsubscribed. Our tests seems to indicate that everything is working correctly and no other people are reporting this issue.

A custom segment normally doesn't do unsubscribes.

Perhaps you're handling one of the events wrong? I think it's best to put some logging around the code that unsubscribes users and use the stack trace to pinpoint the source of the call.

**Comment by freekmurze** [comment]
Closing this for now, as I can't see what the package does wrong. If you have any more info, let us know.

**Comment by ziming** [comment]
ok, here are some updates. So it looks like their email client automatically tries to check each link and hence triggered the unsubscribe link every time i sent to them as well.

As for why this doesn't happen back in the days of using Mailchimp, I suppose Mailchimp either has an extra confirmation page (or ask why you unsubscribing screen first) or that mailchimp is famous enough that the email client knows not to hit the unsubscribe url. I intentionally triggered some errors on the unsubscribed event and check in sentry that a unsubscribe url endpoint is being hit too.

Now a second issue however. Each time they auto unsubscribe, it's not a campaign unsubscribe but an unsubcribe from the whole email list. I believe this is not correct since when you click on the unsubscribe link, a campaign unsubscribe record should be created.
